zcash_primitives/src/sapling.rs

```diff
@@ -185,15 +185,36 @@ impl ProofGenerationKey {
     pub fn to_viewing_key(&self) -> ViewingKey {
         ViewingKey {
             ak: self.ak,
-            nk: constants::PROOF_GENERATION_KEY_GENERATOR * self.nsk,
+            nk: NullifierDerivingKey(constants::PROOF_GENERATION_KEY_GENERATOR * self.nsk),
         }
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct NullifierDerivingKey(jubjub::SubgroupPoint);
+
+impl From<jubjub::SubgroupPoint> for NullifierDerivingKey {
+    fn from(p: jubjub::SubgroupPoint) -> Self {
+        NullifierDerivingKey(p)
+    }
+}
```

Comment:
Hmm.. _do_ we want to expose this capability (remembering that `From` impls will _always_ be public and can't be scoped down)? Where would it be used (outside of this crate's code where we ensure it is constructed correctly from the FVK)?

The only rationale I can see for having either `From` impl is to preserve the current API of `ViewingKey` (where it can be constructed from arbitrary `jubjub::SubgroupPoint`s, but that's not something I want to allow long-term).